.editorconfig

```diff
@@ -0,0 +1,21 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
+tab_width = 4
+trim_trailing_whitespace = true
+
+[*.py]
+max_line_length = 101
```

Comment:
@danimtb suggested to change the line length. Maybe we can make this a bit longer? 🤔

Reply:
Yes, I think it will be more convenient for this repo to have a longer line length. @Croydon what do you think it will be a good default?

Reply:
I have written a small script to figure out the current longest lines in Python files in CCI: https://gist.github.com/Croydon/7ff833b41e9ee48270147e3615510fe6

Biggest line with 273 characters is currently libmysqlclient with this line https://github.com/conan-io/conan-center-index/blob/dff8f07a362ceea6c90381062ca443793ec1ae9b/recipes/libmysqlclient/all/conanfile.py#L50

but this line can be made easily smaller by introducing line breaks after each function argument.

The second biggest line with 217 characters is libpng here: https://github.com/conan-io/conan-center-index/blob/dff8f07a362ceea6c90381062ca443793ec1ae9b/recipes/libpng/all/conanfile.py#L49

This could be reduced by reducing the indentation by making the first line break within `replace_in_file` before the first function argument (instead of after).

Third is pcre with 214 in https://github.com/conan-io/conan-center-index/blob/dff8f07a362ceea6c90381062ca443793ec1ae9b/recipes/pcre/all/conanfile.py#L25

the `default_options` should be split to one per line

Then it is `xz_utils` with 202 characters with another [tools.replace_in_file()](https://github.com/conan-io/conan-center-index/blob/dff8f07a362ceea6c90381062ca443793ec1ae9b/recipes/xz_utils/all/conanfile.py#L69) - again line break can happen before the first argument, making the indentation smaller

After this we ending up again with the same libpng line as above, which I have modified for testing purposes, which would have now 199 characters.

Further modifications might hurt the readability of the code more than the length of the line.
And I think that it is clear that in most cases the longest lines are coming from `tools.replace_in_file()` calls.

If we want to start rather relaxed with line lengths (specifically when this should get enforced at some point), then 200 characters might be a start.

It would still prevent these very extreme cases mentioned above, without being to annoying for contributors who already have to pass an increasing amount of hooks 😄

Reply:
Wow! Thanks a lot @Croydon!! Let's got for those 200 chars. @madebr could you please update that? 😄

Reply:
I've made the change, but with protest.
I believe that shorter lines are easier to read and comprehend, and lead to better code.
Not because you can, you should. :smile:

Reply:
We will continue iterating this file with the consensus of the community as we did here. Thanks a lot anyway for bringing our attention to this useful feature!